api/host_authentication.py

```diff
@@ -19,3 +29,24 @@ def authenticate(self, request):
             raise exceptions.AuthenticationFailed('No such user')
 
         return (user, None)
+
+
+class SignatureValidation(authentication.BaseAuthentication):
+    def authenticate(self, request):
+        signature = request.META.get("HTTP_X_WC_WEBHOOK_SIGNATURE")
+        if not signature:
+            return None
+
+        generated_signature = generate_woocommerce_signature(
+            request.body, os.getenv("WC_WEBHOOK_SECRET")
+        )
+
+        if signature != generated_signature:
+            raise exceptions.ValidationError("Invalid payload checksum signature")
+
+        try:
+            user = User.objects.get(username="pau")
```

Comment:
I think it would be good to parameterized the user while we're at it, even though we might still rely on `pau`.